### PR TITLE
Fixes retina issues on the mac

### PR DESCRIPTION
**Pull request by antiboredom**
This also includes the framerate fix by @parsoyaarihant.

### CODE REVIEW

**Comment by Manindra29** [comment]
I tried it on my machine today. I ran a bunch of sketches in the examples. The frame rates look much better! This is great @antiboredom!

But I did notice some screen tearing. For example in [interpolate.py](https://github.com/p5py/p5-examples/blob/master/basics/math/interpolate.py). Did anyone else notice the same? Not sure if this PR necessarily causes it, because frame rates were too low on Mac for the tearing to be noticeable without this PR.

**Comment by abhikpal** [comment]
> But I did notice some screen tearing. For example in interpolate.py. Did anyone else notice the same?

I don't notice any tearing. That can be worked on later, though. At least the sketches now work on Mac :)

File: p5/sketch/base.py [hunk 1]
```diff
@@ -62,6 +56,10 @@
     config=config,
 )
 
+actual_width, actual_height = window.get_viewport_size()
```
Comment (abhikpal) [inline p5/sketch/base.py]: This seems to break the documentation auto-generation (see #50) I've pushed a fix on the ```fix-50-doc-autogen``` branch, can you see if it still works as expected on the Mac?

**Comment by abhikpal** [comment]
@Manindra29, @antiboredom, could you try testing out the fix in the documentation generation (see #50) in ```fix-50-doc-autogen``` on your respective computers?

**Comment by Manindra29** [comment]
The documentation build passed with #50 on my machine.

**Comment by antiboredom** [comment]
@abhikpal I wasn't able to build the docs, but I think it's related to me messing up my python installation. That said everything else seems to work...

**Pull request merged**